src/DiffWindow.ts

```diff
@@ -0,0 +1,87 @@
+import JoplinData from "api/JoplinData";
+import JoplinViewsDialogs from "api/JoplinViewsDialogs";
+
+export class DiffWindow {
+    
+    private dialogsApi : JoplinViewsDialogs;
+    private dataApi : JoplinData;
+    private installDir : string;
+    private fileSys;
+    
+    constructor(dilg : JoplinViewsDialogs, data : JoplinData, fileSys, installDir : string) {
+        this.dialogsApi = dilg;
+        this.dataApi = data;
+        this.installDir = installDir;
+        this.fileSys = fileSys;
+    }
+
+
+    private handle: string = "";
+    
+    public async Init(id : string) {
+        this.handle = await this.dialogsApi.create(id);
+		await this.dialogsApi.addScript(this.handle, './lib/codemirror/lib/codemirror.js');
+		await this.dialogsApi.addScript(this.handle, './lib/codemirror/lib/codemirror.css');
+		await this.dialogsApi.addScript(this.handle, './lib/diff_match_patch/diff_match_patch.js');
+		await this.dialogsApi.addScript(this.handle, './lib/codemirror/addon/merge/merge.css');
+		await this.dialogsApi.addScript(this.handle, './ui/DiffWindow/index.js');
+		await this.dialogsApi.addScript(this.handle, './ui/DiffWindow/index.css');
+		await this.dialogsApi.setButtons(this.handle, [
+			{
+				id: 'submit',
+				title: 'Save'
+			},
+			{
+				id: 'cancel',
+				title: 'Cancel'
+			}
+		]);
+    }
+
+    public async OpenWindow(noteId: string, compareWithId : string = "") {
+        const localNote = await this.dataApi.get(['notes', noteId], {
+            fields: ['is_conflict', 'conflict_original_id', 'body', 'title']
+        });
+    
+        if(localNote.is_conflict === 0) {
+            throw new Error('This is not a conflict note.');
+        }
+    
+        if(localNote.conflict_original_id.length === 0 && compareWithId == "") {
+            throw new Error('A note must be specificed to be compared with in case of missing original id.');
+        }
+    
+        const remoteId = (compareWithId == "" ? localNote.conflict_original_id : compareWithId);
+
+        const remoteNote = await this.dataApi.get(['notes', remoteId], {
+            fields: ['body', 'title']
+        });
+    
+        const remoteNoteContent = remoteNote.body.replace(/"/g, '&quot;');
+        const localNoteContent = localNote.body.replace(/"/g, '&quot;');
+        const remoteNoteTitle = remoteNote.title.replace(/"/g, '&quot;');
+        const localNoteTitle = localNote.title.replace(/"/g, '&quot;');
+    
+        const htmlContents = await new Promise((res, rej) => {
+            this.fileSys.readFile(this.installDir + '/ui/DiffWindow/index.html', (err: Error, data: string) => {
```

Comment:
I don't think you need this. File operations return promises so you could just do `const htmlContents = await this.fileSys.readFile...`